### ISSUE TITLE
Move NewsWeb to Javascript

### ISSUE THREAD

**Issue by MrEricSir** [MrEricSir/Fang#28]
While investigating ticket #26, I found out that the QML 2 WebView no longer has an easy way to access its DOM in C++.  This is because the QML layer is no longer just a thin wrapper around QtWebKit, but has its own model due to changes in WebKit 2.

So the solution is (unfortunately) to enable JavaScript and pass NewsItems back and forth that way.  The NewsWeb C++ layer will be a thinner layer that simply loads data and writes the bookmark status.

The Javascript code will need to:
- Expose a function to append a NewsItem
- Handle bookmarking items as they go off screen
- Handle endless scrolling: see ticket #25 

The new version of NewsWeb will need to:
- Provide NewsItems as requested
- Read/write bookmark status

Both layers will communicate via QML, similar to how NewsWeb works today.

The HTML cleanup code can be merged into the rest of the pre-insertion cleanup code.

This has to be done prior to the switch to QML 2 as explained in ticket #26
